drivers/gpu/drm/panel/panel-simple.c

```diff
@@ -4681,23 +4684,123 @@ static const struct of_device_id dsi_of_match[] = {
 	}, {
 		.compatible = "osddisplays,osd101t2045-53ts",
 		.data = &osd101t2045_53ts
+	}, {
+		/* Must be the last entry */
+		.compatible = "panel-dsi",
+		.data = &panel_dsi,
 	}, {
 		/* sentinel */
 	}
 };
 MODULE_DEVICE_TABLE(of, dsi_of_match);
 
+
+/* Checks for DSI panel definition in device-tree, analog to panel_dpi */
+static int panel_dsi_dt_probe(struct device *dev,
+			  struct panel_desc_dsi *desc_dsi)
+{
+	struct panel_desc *desc;
+	struct display_timing *timing;
+	const struct device_node *np;
+	const char *dsi_color_format;
+	const char *dsi_mode_flags;
+	int ret;
+
+	np = dev->of_node;
+
+	desc = devm_kzalloc(dev, sizeof(*desc), GFP_KERNEL);
+	if (!desc)
+		return -ENOMEM;
+
+	timing = devm_kzalloc(dev, sizeof(*timing), GFP_KERNEL);
+	if (!timing)
+		return -ENOMEM;
+
+	ret = of_get_display_timing(np, "panel-timing", timing);
+	if (ret < 0) {
+		dev_err(dev, "%pOF: no panel-timing node found for \"panel-dsi\" binding\n",
+			np);
+		return ret;
+	}
+
+	desc->timings = timing;
+	desc->num_timings = 1;
+
+	of_property_read_u32(np, "width-mm", &desc->size.width);
+	of_property_read_u32(np, "height-mm", &desc->size.height);
+
+	of_property_read_u32(np, "lanes", &desc_dsi->lanes);
+
+	of_property_read_string(np, "dsi-color-format", &dsi_color_format);
+	if (!strcmp(dsi_color_format, "RGB888")) {
+		desc_dsi->format = MIPI_DSI_FMT_RGB888;
+		desc->bpc = 8;
+	} else if (!strcmp(dsi_color_format, "RGB565")) {
+		desc_dsi->format = MIPI_DSI_FMT_RGB565;
+		desc->bpc = 6;
+	} else if (!strcmp(dsi_color_format, "RGB666")) {
+		desc_dsi->format = MIPI_DSI_FMT_RGB666;
+		desc->bpc = 6;
+	} else if (!strcmp(dsi_color_format, "RGB666_PACKED")) {
+		desc_dsi->format = MIPI_DSI_FMT_RGB666_PACKED;
+		desc->bpc = 6;
+	}
+
+	of_property_read_string(np, "mode", &dsi_mode_flags);
+	if (!strcmp(dsi_mode_flags, "MODE_VIDEO"))
```

Comment:
The MIPI_DSI_MODE_xxx flags are generally used in combination, not mutually exclusive.
eg MODE_VIDEO switches between command mode and video mode displays. Almost all the others are only applicable to video mode, so have to have that set as well.

If your example binding with
```
mode = "MODE_VIDEO", "MODE_VIDEO_BURST", "MODE_NO_EOT_PACKET";
```
is valid, then that is great. However I don't see this parsing code reading all of those flags.

Reply:
Hint: Look at `of_property_for_each_strings`, which uses the length of the property to decide if there may be more strings.

Aside: I find the following related code in `of_prop_next_string` worrying:
```
	curv += strlen(cur) + 1;
	if (curv >= prop->value + prop->length)
		return NULL;
```
The second line acknowledges that the last string may not be NUL-terminated by comparing the apparent length of the string with the overall length of the property, but by then it may be too late - the strlen may already have stumbled into an invalid page. I should work up the enthusiasm to raise the matter upstream.

Reply:
Ooh wow yea that fell of my todo, thanks for spotting that. My panel for testing only requires MODE_VIDEO hence working fine despite this bug..